### ISSUE TITLE
Well known proto3 types missing from npm package

### ISSUE THREAD

**Issue by KeithWoods** [google/protobuf#1638]
Hi All

I'm using proto3 with the last beta 3 (v3.0.0-beta-3) to generate javascript DTOS as CommonJs modules.

In the generated javascript, built in proto type (e.g. `google.protobuf.Timestamp`) are 'required' but don't exist.

For example, given this proto:
```
// file: foo.proto
syntax = "proto3";

import "google/protobuf/timestamp.proto";

message SomeMessage {
   google.protobuf.Timestamp someTime = 1;
}
```

And this command:
```
protoc --proto_path=./ --js_out=import_style=commonjs:./ foo.proto
```
we get this output: 

```
// file: foo_pb.js
/**
* @fileoverview
* @enhanceable
* @public
*/
// GENERATED CODE -- DO NOT EDIT!

var jspb = require('google-protobuf');
var goog = jspb;
var global = Function('return this')();

var google_protobuf_timestamp_pb = require('./google/protobuf/timestamp_pb.js'); // NOTE ./google/protobuf/timestamp_pb.js doesn't exist
goog.exportSymbol('proto.SomeMessage', null, global);
```

The file `timestamp_pb.js` isn't generated (note `timestamp.proto` does exist under my protoc executable) thus doesn't exist at the generated path.

I'd expect it to exist in the `google-protobuf` npm package, e.g. the generated code should be `require('google-protobuf/somePathHere/timestamp_pb.js')`.

First [reported on google groups](https://groups.google.com/forum/#!topic/protobuf/46NJpIX-kFI), it was suggested I create this issue.

**Comment by haberman** [comment]
The well-known types are in the npm package now.

**Comment by thefunnyguy** [comment]
When I use NPM package to get well know know types, it gives the file with import style as commonjs.

How do I get well know types for import style closure?

**Comment by acozzette** [comment]
@thefunnyguy  I think for now you just have to build the well-known types manually with protoc if you want to use Closure-style imports. We could potentially consider including both import styles in the NPM package, though. What does your build setup look like?

**Comment by thefunnyguy** [comment]
@acozzette 

Some of my proto files are using "import "google/protobuf/timestamp.proto";" 

I am using bazel build system so all code is closure style. Initially I tried using "closure_js_proto_library"        rule to covert my protos to closure style. But well known types are not available, so its failing.

I ended up running the below command to generate JS file:
protoc --proto_path=./proto-folder/ --js_out=library=proto_lib,binary:. ./proto-folder/*.proto

But still goog.require for well known type fails as those closure style files are not available.

I thought to get those file using NPM, but the files are in commonjs style, so google require statements are still failing.

I ended up taking the whole google protobuf folder and compiled the proto files using protoc, something like:
proto_path=./protobuf/src/google/protobuf/ --js_out=library=well_known_proto_libs,binary:. ./protobuf/src/google/protobuf/timestamp.proto

This seems to resolve the errors for now, but not sure if that's the correct way. Having closure style NPM package will make life easy i guess.
